VirtoCommerce.PricingModule.Data/Services/PricingServiceImpl.cs

```diff
@@ -426,10 +487,28 @@ public virtual void DeletePricelistsAssignments(string[] ids)
                 CommitChanges(repository);
                 ResetCache();
             }
-        }      
+        }
         #endregion
 
+        protected virtual IEnumerable<coreModel.Price> FilterTimeOverlappingPrices(IEnumerable<coreModel.Price> prices, DateTime certainDate)
+        {
+            if (prices == null) yield break;
 
+            foreach (var pricesForProduct in prices
+                .Where(x => x != null)
+                .GroupBy(x => x.ProductId))
+            {
+                var orderedPrices = pricesForProduct
+                    // Startdate closest to certainDate is more important.
+                    .OrderBy(x => certainDate.Subtract(x.StartDate.GetValueOrDefault()).TotalSeconds)
+                    // Enddate closest to certainDate is more important.
+                    .ThenBy(x => x.EndDate.GetValueOrDefault().Subtract(certainDate).TotalSeconds);
+
+                var filteredPrice = orderedPrices.FirstOrDefault();
+                if (filteredPrice != null)
```

Comment:
this condition is redundant you might simplify this `yeild orderedPrices.First()`

Reply:
Right, fixed.